File: 026.py
import math


def is_prime(num):
	if num == 1:
		return False

	s = int(math.sqrt(num))
	if (num % 2 == 0 and 2 <= s) or (num % 3 == 0 and 3 <= s):
		return False

	for p in range(5, s + 1, 6):
		if num % p == 0 or num % (p + 2) == 0:
			return False

	return True


def order(a, n):
	"""order of a mod n"""
	k = 1
	while a**k % n != 1:
		#print a, n, k
		k += 1
	return k


def main():
	longest = 1
	n = 0
	# Only consider the set of prime numbers
	for d in range(11, 1000, 6) + range(7, 1000, 6):
		if not is_prime(d):
			continue
		l = order(10, d)
		if l > longest:
			longest = l
			n = d
	print n


if __name__ == '__main__':
	main()

File: 026.cpp
#include <iostream>
#include <cstdint>
#include "euler.hpp"


int main(void)
{
	uint16_t longest = 0;
	uint16_t n = 0, d, ord;

	for (d = 7; d < 1000; d += (d % 6 == 5) ? 2 : 4) {
		if (!euler::is_prime(d))
			continue;
		ord = euler::order(10, d);
		if (ord > longest) {
			longest = ord;
			n = d;
		}
	}

	std::cout << n << std::endl;

	return 0;
}
